pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -1306,3 +1313,102 @@ func (vs *VSphere) NodeManager() (nodeManager *NodeManager) {
 	}
 	return vs.nodeManager
 }
+
+func withTagsClient(ctx context.Context, connection *vclib.VSphereConnection, f func(c *tags.RestClient) error) error {
+	vsURL := connection.Client.URL()
+	vsURL.User = url.UserPassword(connection.Username, connection.Password)
+	c := tags.NewClient(vsURL, connection.Insecure, "")
+	if err := c.Login(ctx); err != nil {
+		return err
+	}
+	defer c.Logout(ctx)
+	return f(c)
+}
+
+// GetZone implements Zones.GetZone
+func (vs *VSphere) GetZone(ctx context.Context) (cloudprovider.Zone, error) {
+	nodeName, err := vs.CurrentNodeName(ctx, vs.hostName)
+	if err != nil {
+		glog.Errorf("Cannot get node name.")
+		return cloudprovider.Zone{}, err
+	}
+	zone := cloudprovider.Zone{}
+	vsi, err := vs.getVSphereInstanceForServer(vs.cfg.Workspace.VCenterIP, ctx)
+	if err != nil {
+		glog.Errorf("Cannot connent to vsphere. Get zone for node %s error", nodeName)
+		return cloudprovider.Zone{}, err
+	}
+	dc, err := vclib.GetDatacenter(ctx, vsi.conn, vs.cfg.Workspace.Datacenter)
+	if err != nil {
+		glog.Errorf("Cannot connent to datacenter. Get zone for node %s error", nodeName)
+		return cloudprovider.Zone{}, err
+	}
+	vmHost, err := dc.GetHostByVMUUID(ctx, vs.vmUUID)
+	if err != nil {
+		glog.Errorf("Cannot find VM runtime host. Get zone for node %s error", nodeName)
+		return cloudprovider.Zone{}, err
+	}
+	client := vsi.conn
+	err = withTagsClient(ctx, client, func(client *tags.RestClient) error {
+		tags, err := client.ListAttachedTags(
```

Comment:
This a short parameter list, no need for multiple lines: `client.ListAttachedTags(ctx, vmHost)`